compiler/src/java_plugin/cpp/java_generator.cpp

```diff
@@ -631,6 +654,9 @@ static void PrintStub(
 
   // RPC methods
   for (int i = 0; i < service->method_count(); ++i) {
+    if (impl_base) {
```

Comment:
Instead of skipping the body of the 100 line loop, why not get rid get rid of the `PrintStub(service, vars, p, ABSTRACT_CLASS);` call entirely and "do it manually"? The main point of `PrintStub()` is to loop over the methods, which doesn't need to happen. Seems this would be equivalent:

```java
  GrpcWriteServiceDocComment(p, service, type);
  if (service->options().deprecated()) {
    p->Print(*vars, "@$Deprecated$\n");
  }
  p->Print(
      *vars,
      "public static abstract class $service_name$ImplBase\n"
      "    implements $BindableService$, AsyncService {}\n");
```

Reply:
There is actually a bindService method defined in this class, but it does make sense to split it out as nearly all of the logic is separate from the other stubs'.